dgraph/graphql/src/dgraph/queries.js

```diff
@@ -0,0 +1,19 @@
+export const queries = {
+  entities: (filter) => {
+    const type = filter?.type ?? "medicinal_product";
+    const codeFilter = filter?.code ? `@filter(eq(code, "${filter.code}"))` : ``;
+
+    return `{
+      query(func: eq(type, ${type})) ${codeFilter} @recurse(loop: false)  {
+        code
+        description
+        type
+        value
+        has_child
+        has_property
+      }
+    }`
+  }
+}
```

Comment:
I think a few things here:

1. `code` is always unique, so can be a different query.
2. Let the resolver parse the args (rather than one big `entities` method which does everything). We will need to have a think about how to handle all the different filtering possibilities without combinatoric blowout).

My suggestion:

```suggestion
export const queries = {
  entity: code => // get entity by code
  entities: type => // get entities by type
}
```

See below for resolver suggestion,

Reply:
Cool, yeah I was thinking along the same lines after realising the inadvertent bug 😸 . Good point r.e. moving args into resolvers, will do.

Also realised there's a bit of a dodgy bug where the current Dgraph query lets you query for a type of 'strength' and returns properties instead of entities (which then confuses Apollo), will fix this too.

Reply:
Fixes applied in https://github.com/openmsupply/unified-codes/commit/b5eebe44775bff017d8b236cbf95e78afbcee3ac